examples/units_and_coordinates/spice.py

```diff
@@ -0,0 +1,142 @@
+"""
+============================================
+Coordinates computations using SPICE kernels
+============================================
+
+How to use SPICE kernels provided by space missions to perform coordinates
+computations
+
+The `SPICE <https://naif.jpl.nasa.gov/naif/>`__ observation geometry information
+system is being increasingly used by space missions to describe the locations of
+spacecraft and the time-varying orientations of reference frames.  Here are two
+examples of mission SPICE kernels:
+
+* `Solar Orbiter <https://www.cosmos.esa.int/web/spice/solar_orbiter>`__
+* `Parker Solar Probe <https://sppgway.jhuapl.edu/ancil_products>`__
+
+The `sunpy.coordinates.spice` module enables the use of the
+`~astropy.coordinates.SkyCoord` to API to perform SPICE computations such as the
+location of bodies or the transformation of a vector from one coordinate frame
+to another coordinate frame.  Although SPICE kernels can define coordinate
+frames that are very similar to the frames that `sunpy.coordinates` already
+provides, there will very likely be slight differences.  Using
+`sunpy.coordinates.spice` will ensure that the definitions are exactly what the
+mission specifies and that the results are identical to other implementations
+of SPICE (e.g., CSPICE or Icy).
+
+.. note::
+    This example requires the optional dependency `~spiceypy.spiceypy` to be
+    installed.
+
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.dates import DateFormatter
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+from sunpy.coordinates import spice
+from sunpy.data import cache
+from sunpy.time import parse_time
+
+###############################################################################
+# Download a subset (~30 MB) of the Solar Orbiter SPICE kernel set that
+# corresponds to about a day of the mission.
+
+obstime = parse_time('2022-10-12') + np.arange(720) * u.min
+
+kernel_urls = [
```

Comment:
We clearly need some functionality for getting kernels. Is the functionality in astrospice capable of doing this?

Reply:
`astrospice` has some limited capability of doing this, but of course it's at a very early stage of development.  For example, for Solar Orbiter, `astrospice` downloads only the SPK kernel for spacecraft location, and only the predicted kernel at that.  In order to support all things SPICE, `astrospice` would need to be fleshed out to download all of the different types of kernels.

Reply:
yea code to download/manage the kernels would be super handy for example give a SO spice repo and meta kernel path code could download and load the spice kernels